app/src/main/java/com/tari/android/wallet/ui/fragment/onboarding/LocalAuthFragment.kt

```diff
@@ -0,0 +1,353 @@
+    lateinit var buttonAuthFormat: String
+    @BindString(R.string.auth_prompt_touch_id)
+    lateinit var authTouchId: String
+    @BindString(R.string.auth_prompt_pin)
+    lateinit var authPin: String
+    @BindString(R.string.auth_prompt_touch_id_desc)
+    lateinit var authTouchIdDesc: String
+    @BindString(R.string.auth_prompt_pin_desc)
+    lateinit var authPinDesc: String
+
+    private var authType: AuthType = AuthType.None
+
+    private var listener: Listener? = null
+
+    override val contentViewId = R.layout.fragment_local_auth
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        if (context is Listener) {
+            listener = context
+        } else {
+            throw AssertionError("Activity must implement listener")
+        }
+    }
+
+    private fun setDeviceAuthType() {
+        authType = when {
+            BiometricManager.from(context!!).canAuthenticate() == BIOMETRIC_SUCCESS -> {
+                AuthType.Biometric
+            }
+            AuthUtil.isDeviceSecured(context!!) -> {
+                AuthType.Pin
+            }
+            else -> {
+                AuthType.None
+            }
+        }
+    }
+
+    private fun setupUi() {
+        if (authType == AuthType.Biometric) {
+            //setup ui for fingerprint auth
+            titleAuthTypeTextView.text = authTouchId
+            useAuthButton.text = String.format(buttonAuthFormat, authTouchId)
+            authDescTextView.text = authTouchIdDesc
+            authTypeImageView.setImageResource(R.drawable.fingerprint)
+            authSuccessAuthTypeImageView.setImageResource(R.drawable.fingerprint)
+            authSuccessAuthTypeTitleTextView.text = authTouchId
+        } else {
+            //setup ui for pin or password auth
+            titleAuthTypeTextView.text = authPin
+            useAuthButton.text = String.format(buttonAuthFormat, authPin)
+            authDescTextView.text = authPinDesc
+            authTypeImageView.setImageResource(R.drawable.numpad)
+            authSuccessAuthTypeImageView.setImageResource(R.drawable.numpad)
+            authSuccessAuthTypeTitleTextView.text = authPin
+        }
+
+        if (authType == AuthType.None) {
+            authFailed()
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setDeviceAuthType()
+        setupUi()
+
+        rootView.viewTreeObserver.addOnGlobalLayoutListener(
+            object : ViewTreeObserver.OnGlobalLayoutListener {
+                override fun onGlobalLayout() {
+                    rootView.viewTreeObserver.removeOnGlobalLayoutListener(this)
+                    playStartUpAnim()
+                }
+            })
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        listener = null
+    }
+
+    @OnClick(R.id.local_auth_btn_use_auth)
+    fun onAuthClick() {
```

Comment:
Let's make the id's and click action functions follow what's written on the button. Maybe more like `onEnablePINButtonClick`/`onEnableAuthButtonClick`, or change the text on the button to go "Authenticate".